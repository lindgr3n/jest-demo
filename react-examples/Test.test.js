import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer';
import {shallow, mount} from 'enzyme'

import TestComponent from './Test'

describe('<Test>', () => {
    /*
    beforeEach(() => {
        initializeDatabase();
    });

    afterEach(() => {
        clearDatabase();
    });
    */

    it('Should render component', () => {
        const div = document.createElement('div');
        ReactDOM.render(<TestComponent />, div);
    })

    it('Snapshottesting', () => {
        const tree = renderer.create(<TestComponent />).toJSON();
        expect(tree).toMatchSnapshot();
    })

    it('Snapshottesting with enzyme', () => {
        const result = shallow(<TestComponent />)
        expect(result).toMatchSnapshot()
    })

    it('Check correct value for item', () => {
        const result = shallow(<TestComponent />)
        // Works like queryselector
        const thirdItem = result.find('.icon-test-3')
        expect(thirdItem.text()).toEqual('Three')
    })

    it('Should contain three <i> elements', () => {
        const result = shallow(<TestComponent />)
        const iElements = result.find('i')
        expect(iElements.length).toEqual(3)
    })

    it('Snapshot should match number of elements', () => {
        const result = shallow(<TestComponent />)
        const iElements = result.find('i')
        expect(iElements.length).toMatchSnapshot()
    })
}) 