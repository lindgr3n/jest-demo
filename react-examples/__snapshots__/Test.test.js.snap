// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Test> Snapshot should match number of elements 1`] = `3`;

exports[`<Test> Snapshottesting 1`] = `
<div
  className="my-component"
>
  <i
    className="icon-test-1"
  >
    One
  </i>
  <br />
  <i
    className="icon-test-2"
  >
    Two
  </i>
  <br />
  <i
    className="icon-test-3"
  >
    Three
  </i>
</div>
`;

exports[`<Test> Snapshottesting with enzyme 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="my-component"
>
    <i
        className="icon-test-1"
    >
        One
    </i>
    <br />
    <i
        className="icon-test-2"
    >
        Two
    </i>
    <br />
    <i
        className="icon-test-3"
    >
        Three
    </i>
</div>,
  "nodes": Array [
    <div
      className="my-component"
>
      <i
            className="icon-test-1"
      >
            One
      </i>
      <br />
      <i
            className="icon-test-2"
      >
            Two
      </i>
      <br />
      <i
            className="icon-test-3"
      >
            Three
      </i>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Test />,
      "_debugID": 18,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Test {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="my-component"
>
          <i
                    className="icon-test-1"
          >
                    One
          </i>
          <br />
          <i
                    className="icon-test-2"
          >
                    Two
          </i>
          <br />
          <i
                    className="icon-test-3"
          >
                    Three
          </i>
</div>,
        "_debugID": 19,
        "_renderedOutput": <div
          className="my-component"
>
          <i
                    className="icon-test-1"
          >
                    One
          </i>
          <br />
          <i
                    className="icon-test-2"
          >
                    Two
          </i>
          <br />
          <i
                    className="icon-test-3"
          >
                    Three
          </i>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Test />,
}
`;
